server/src/node/main.ts

```diff
@@ -26,8 +27,60 @@ export namespace Files {
 	export const resolveModulePath = fm.resolveModulePath;
 }
 
+let _isDetached: boolean | undefined = undefined;
+
+/**
+ * @returns {boolean} True if the process is detached from its parent, false otherwise.
+ *
+ * @description
+ * When a process is detached, it will continue running even if it's parent process
+ * terminates. **But EXIT notifications from the client will be honored regardless.**
+ *
+ * @remarks
+ * This function assumes the parent process has been properly configured with
+ * necessary settings (e.g., using `child_process.unref()`).
+ */
+function isDetached() {
+	// cached result
+	if (_isDetached !== undefined) {
+		return _isDetached;
+	}
+
+	const args = parseCliOpts(process.argv);
+
+	if (Object.keys(args).includes('detached')) {
+		_isDetached = true;
+
+		// Override the default behavior: when then parent/child communication
+		// channel (e.g. IPC) disconnects, the child terminates.
+		process.on('disconnect', () => {
+			endProtocolConnection();
+		});
+
+		const timeoutValue = args['detached'];
+		const timeoutMillis = Number(timeoutValue);
+		if (timeoutValue !== undefined && isNaN(timeoutMillis)) {
+			throw Error(`Cli value for flag '--detached' was not a number: ${timeoutMillis}`);
+		}
+		// Keeps the server alive since node may terminate this process if
+		// the parent terminates + this process has nothing in the event loop.
+		setInterval(() => {
+			// Check if the detached server has reached the user-specified lifetime
+			if (_protocolConnectionEndTime && timeoutMillis && Date.now() - _protocolConnectionEndTime >= timeoutMillis) {
```

Comment:
Although not happening here I usually try to make undefined checks explicit especially if the variable as a or type that can falsify. 

`_protocolConnectionEndTime !== undefined`